packages/integration-sdk-runtime/src/storage/FileSystemGraphObjectStore/FileSystemGraphObjectStore.ts

```diff
@@ -166,7 +222,7 @@ export class FileSystemGraphObjectStore implements GraphObjectStore {
    * and prematurely end, causing the next step to start up before the
    * data it depends on is present on disk.
    */
-  async lockOperation(operation: () => Promise<any>) {
+  private async lockOperation<T>(operation: () => Promise<T>) {
```

Comment:
Looking at the [documentation](https://www.npmjs.com/package/p-map) for `p-map`, it looks there is a concurrency parameter that we can use instead of using a semaphore.

Reply:
Ya, I think there are ways we can get rid of this, but we should tackle it separately.